.github/workflows/master-build.yml

```diff
@@ -0,0 +1,36 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: master pr build
+
+on: [push, pull_request]
```

Comment:
Hi @madhawa-gunasekara, the check runs well. But When i tried it in own repo [here ](https://github.com/balashashanka/nutch/tree/NUTCH-2181) and [here](https://github.com/balashashanka/nutch/pull/1). For a PR same check is ran twice. So may be run this only for PR or split this to run check on master after merge and for a PR.

Reply:
Hi @balashashanka 
Here you can see, I added workflows for both pull requests and direct pushes. so basically It will run 1 time for the pull request, and one time after the merge. (push). I haven't mentioned a specific branch for pushes. so this will enable for every branch. [In Gora](https://github.com/apache/gora/blob/master/.github/workflows/master-push-build.yml#L22), we have specified master branch for pushes. Let me know if we need to specify particular branches for pushes.

Reply:
Yes it would be nice if we add it to master as it will be triggered only once for every pull request.

Reply:
Don't you need for 2.x branch?

Reply:
Since we dont have an official dev support on 2.x I dont think we would need it. 
But @sebastian-nagel or @lewismc can give a bit of clarity on this.
Or in the mean time we can create a different PR with file for 2.x and merge if its required.

Reply:
Agreed, 2.x branch development was stopped last year and I don't think we're going to accept any contributions to that branch on the foreseeable future.

Reply:
Okay, I updated